src/main/java/it/gov/pagopa/pu/debtpositions/service/DebtPositionServiceImpl.java

```diff
@@ -93,5 +97,16 @@ public PagedDebtPositions getPagedDebtPositionsByIngestionFlowFileId(Long ingest
   public void delete(DebtPosition debtPosition) {
     debtPositionDeleteService.delete(debtPosition);
   }
+
+  @Override
+  public Optional<DebtPosition> getDebtPositionByIupdAndOrganizationId(String iupd, Long organizationId) {
+    return debtPositionRepository.findDebtPositionByIupdOrgAndOrganizationId(iupd, organizationId);
+  }
+
+  @Override
+  public void updateDebtPosition(Long debtPositionId, DebtPositionDTO debtPositionDTO) {
+    DebtPosition debtPosition = debtPositionRepository.findById(debtPositionId).orElseThrow(() -> new EntityNotFoundException("debtPositionId=" + debtPositionId + " not found"));
```

Comment:
it would be better to use `findEntityGraphByDebtPositionId` instead of the simple `findById` this way it will fetch the entire hierarchy with just one query